src/glass/src/app/shared/components/datatable/datatable.component.ts

```diff
@@ -221,6 +257,28 @@ export class DatatableComponent implements OnInit, OnDestroy {
     this.reloadData();
   }
 
+  onSelectionChange($event: any, row: DatatableData) {
+    if ($event.target.checked) {
+      this.selected.push(row);
+    } else {
+      const selectedIndex = _.map(this.selected, this.identifier).indexOf(row[this.identifier]);
```

Comment:
You can also use the following code which looks more intuitive IMO.
```suggestion
      const selectedIndex = _.findIndex(this.selected, [this.identifier, row[this.identifier]]);
```